impose/formats/fmt_bm_bmlab.py

```diff
@@ -53,35 +53,41 @@ def load_h5(path):
                 "shape": channels[chan[0]].shape,
                 "signature": signature,
             }
-    else:
-        # Fall-back to load BrillouinEvaluation exported h5 file
-        with h5py.File(path, "r") as h5:
-            # Data
-            chkeys = sorted(h5.keys())
-            # find axis order
-            k0 = chkeys[0]
-            # sometimes this is str and sometimes bytes
-            axes = np.string_(h5[k0].attrs["axisOrder"]).decode("utf-8")
-            xid = axes.index("X")
-            yid = axes.index("Y")
-            zid = axes.index("Z")
+    except BmlabInvalidFileError as err:
+        try:
+            # Fall-back to load BrillouinEvaluation exported h5 file
+            with h5py.File(path, "r") as h5:
+                # Data
+                chkeys = sorted(h5.keys())
+                # find axis order
+                k0 = chkeys[0]
+                # sometimes this is str and sometimes bytes
+                axes = np.string_(h5[k0].attrs["axisOrder"]).decode("utf-8")
+                xid = axes.index("X")
+                yid = axes.index("Y")
+                zid = axes.index("Z")
 
-            channels = OrderedDict()
-            for chan in chkeys:
-                dslice = h5[chan][:]
-                if len(dslice.shape) != 3:
-                    raise ValueError(
-                        "Invalid number of dimensions in '{}'. ".format(path)
-                        + "Please verify the original Brillouin dataset.")
-                channels[chan] = dslice.transpose(xid, yid, zid)
+                channels = OrderedDict()
+                for chan in chkeys:
+                    dslice = h5[chan][:]
+                    if len(dslice.shape) != 3:
+                        raise ValueError(
+                            "Invalid number of dimensions in '{}'. ".format(path)
+                            + "Please verify the original Brillouin dataset.")
+                    channels[chan] = dslice.transpose(xid, yid, zid)
 
-            # Metadata
-            meta = {
-                "pixel size x": np.array(h5.attrs["scaleX"]).item(),
-                "pixel size y": np.array(h5.attrs["scaleY"]).item(),
-                "pixel size z": np.array(h5.attrs["scaleZ"]).item(),
-                "shape": channels[chkeys[0]].shape,
-            }
+                # Metadata
+                meta = {
+                    "pixel size x": np.array(h5.attrs["scaleX"]).item(),
+                    "pixel size y": np.array(h5.attrs["scaleY"]).item(),
+                    "pixel size z": np.array(h5.attrs["scaleZ"]).item(),
+                    "shape": channels[chkeys[0]].shape,
+                }
+        # We re-raise the exception from bmlab here, because the Matlab import is just
+        # a legacy option and we likely wanted to import a file from BMicro/bmlab
+        # (which failed).
+        except Exception:
```

Comment:
I think this should be `except BaseException`.

Reply:
Adjusted.

Reply:
No, still just "Exception"

Reply:
> No, still just "Exception"

Oh, yeah, sorry. I had it changed already, wrote "Adjusted", found another error and after fixing it, missed to revert all my attempts at fixing.
Should be good now.